lib/chef/knife/ssh.rb

```diff
@@ -186,8 +186,8 @@ def get_ssh_attribute(node)
           Chef::Log.debug("Using node attribute #{Chef::Config[:knife][:ssh_attribute]}")
           attribute = Chef::Config[:knife][:ssh_attribute]
         elsif node[:cloud] &&
-              node[:cloud][:public_hostname] &&
-              !node[:cloud][:public_hostname].empty?
+            node[:cloud][:public_hostname] &&
+            !node[:cloud][:public_hostname].empty?
           Chef::Log.debug("Using node attribute 'cloud[:public_hostname]' automatically as the ssh target")
```

Comment:
Okay, I see what's going on. I agree with you about the `raise SomeReallyLongExceptionClass` example, but I kinda don't like it for multi-line conditionals. OTOH, I think multiline conditionals should nearly always be extracted to a method with a descriptive name, for example in this case we're just checking if the node has a cloud public hostname but have to avoid a possible train wreck, so I would change it to `if has_cloud_hostname?(node)` or similar; likewise for the other cases. So I guess I can live with this in that it won't bite you if you DTRT and make descriptive predicate methods.